Userland/Utilities/cat.cpp

```diff
@@ -63,3 +64,10 @@ ErrorOr<int> serenity_main(Main::Arguments arguments)
 
     return 0;
 }
+
+#ifndef EXCLUDE_SERENITY_MAIN
+ErrorOr<int> serenity_main(Main::Arguments arguments)
```

Comment:
It feels very uncomfortable to change 27 of the utilities in the system to use this pattern. Which in general, looks awkward and is not something we would want people copy pasting into new utilities.

What about something like this instead, in the CMakeLists.txt?

```cmake
function (buggiebox_utility src)
    get_filename_component(utility ${src} NAME_WE)
    target_sources(BuggieBox PRIVATE ${src})
    set_source_files_properties(${src} PROPERTIES COMPILE_DEFINITIONS "-Dserenity_main=${utility}_main")
endfunction()

set(utility_srcs
    ../Utilities/cat.cpp
    ../Utilities/checksum.cpp
    ../Utilities/chmod.cpp
    ../Utilities/chown.cpp
    ../Utilities/cp.cpp
    ../Utilities/df.cpp
    ../Utilities/env.cpp
    ../Utilities/file.cpp
    ../Utilities/find.cpp
    ../Utilities/id.cpp
    ../Utilities/less.cpp
    ../Utilities/ln.cpp
    ../Utilities/ls.cpp
    ../Utilities/lsblk.cpp
    ../Utilities/mkdir.cpp
    ../Utilities/mknod.cpp
    ../Utilities/mount.cpp
    ../Utilities/mv.cpp
    ../Utilities/ps.cpp
    ../Utilities/rm.cpp
    ../Utilities/rmdir.cpp
    ../Utilities/tail.cpp
    ../Utilities/tree.cpp
    ../Utilities/umount.cpp
    ../Utilities/uname.cpp
    ../Utilities/uniq.cpp
)

foreach(file IN LISTS utility_srcs)
   buggiebox_utility(file)
endforeach()

target_sources(BuggieBox PRIVATE ../Shell/main.cpp)
set_source_files_properties( ../Shell/main.cpp PROPERTIES COMPILE_DEFINITIONS "-Dserenity_main=shell_main")
```

Reply:
That truly is a :galaxybrain: solution (and something that I completely forgot about). In fact, I would very much prefer it over the solution that I proposed.

Just the name for `sh_main` is wrong, but that's just details at this point. :P

Reply:
This looks cool, still I am concerned about the fact that many utilities use global variables and in the end, we will need the `#ifdef` soup to get rid of those so the BuggieBox will be able to compile (otherwise there will be conflictive global variables in many utilities, which is bad)...

Reply:
>still I am concerned about the fact that many utilities use global variables

How would ifdef soup help with this?

Reply:
The current situation can be solved without using ifdefs all over the place. In #14936, I had to use `#ifdef` a bit more aggressively to get rid of many global variables from some utilities because the compiler found more than one instance for each of them in utilities source files.
So for now, yes, we can go with this route, but what I tried to say that using some `#ifdef`s is unavoidable later on.

Reply:
I mean. Any global variables that utilities are using can simply be converted to file static variables and everything is fine. The only issue there is if we keep the #include foo.cpp pattern. Which I guess is another reason not to use it.

Reply:
Done.